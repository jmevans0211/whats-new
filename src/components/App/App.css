* {
  box-sizing: border-box;
  padding: 0;
}

.app {
  border: black solid 5px;
  border-radius: 10px;
}

.selectedNews {
  display: flex;
  text-align: center;
  width: 100%;
}

img {
  border: black double 3px;
  height: 200px
}
