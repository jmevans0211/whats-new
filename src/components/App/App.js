import React, { Component } from 'react';
import local from '../../data/local';
import NewsContainer from '../NewsContainer/NewsContainer'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      local
    }
  }

  render () {
    return (
      <div className="app">
        <NewsContainer />
      </div>
    );
  }
}

export default App;
