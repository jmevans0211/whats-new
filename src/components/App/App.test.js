import React from 'react';
import { shallow } from 'enzyme';
import entertainment from '../../data/entertainment';
import App from './App';

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when a new topic is selected', () => {
    const wrapper = shallow(<App />);
    const mockNews = {
      entertainment: entertainment
    };
    const expected = mockNews.entertainment;

    wrapper.instance().selectTopic('entertainment')

    expect(wrapper.state('currentTopic')).toEqual(expected);
  });

  it('should update state when something is searched', () => {
    const wrapper = shallow(<App />);
    const currentState = {
      currentTopic: entertainment
    };
    const expected = [
      {
        id: 4,
        headline: 'Metallica Cancel Tour as James Hetfield Returns to Rehab',
        img: 'https://www.rollingstone.com/wp-content/uploads/2019/09/10331207d.jpg?resize=900,600&w=1200',
        description: 'Metallica have postponed their upcoming tour of Australia and New Zealand so James Hetfield could return to rehab, the heavy metal legends announced Friday.',
        url: 'https://www.rollingstone.com/music/music-news/metallica-cancel-tour-james-hetfield-rehab-892158/'
      }
    ]

    wrapper.setState(currentState)

    wrapper.instance().filterSearch('Metallica');

    expect(wrapper.state('currentTopic')).toEqual(expected)
  });
}); //<<<-----end of App



