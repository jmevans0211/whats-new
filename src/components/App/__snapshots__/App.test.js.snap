// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="app"
>
  <SearchForm
    search={[Function]}
  />
  <div
    className="selectedNews"
  >
    <Menu
      selectTopic={[Function]}
    />
  </div>
</div>
`;
